projects/igniteui-angular/src/lib/data-operations/filtering-condition.ts

```diff
@@ -60,8 +60,9 @@ export class IgxFilteringOperand {
     /**
      * @hidden
      */
-    protected findValueInSet(target: any, searchVal: Set<any>) {
-        return searchVal.has(target);
+    protected findValueInSet(target: any, searchVal: Set<any>, ignoreCase?: boolean) {
```

Comment:
This would convert non-string data types to strings as well. I don't think it is a good idea to change the most basic logic for all filtering operands' findValueInSet() method in order to address a very concrete scenario possibly in the ESF only. Please revisit the approach.